src/v/compat/id_allocator_compat.h

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+
+#include "cluster/errc.h"
+#include "cluster/types.h"
+#include "compat/check.h"
+#include "compat/id_allocator_generator.h"
+#include "compat/json.h"
+#include "model/timeout_clock.h"
+
+#include <cstdint>
+
+namespace compat {
+
+/*
+ * cluster::allocate_id_request
+ */
+template<>
+struct compat_check<cluster::allocate_id_request> {
+    static constexpr std::string_view name = "cluster::allocate_id_request";
+
+    static std::vector<cluster::allocate_id_request> create_test_cases() {
+        return compat::generate_instances<cluster::allocate_id_request>();
+    }
+
+    static void to_json(
+      cluster::allocate_id_request obj, json::Writer<json::StringBuffer>& wr) {
+        json::write_member(wr, "timeout", obj.timeout.count());
+    }
+
+    static cluster::allocate_id_request from_json(json::Value& rd) {
+        int64_t count = 0;
+        json::read_member(rd, "timeout", count);
```

Comment:
I also think Brandon has submitted a utility for serializing durations, maybe you can cherry pick his change or just take it and later resolve whoevers PR hits dev second